tests/expectations/test_expectation.py

```diff
@@ -251,3 +257,58 @@ def test_unrecognized_expectation_arg_raises_error():
         gxe.ExpectColumnMaxToBeBetween(
             column="foo", min_value=0, max_value=10, mostyl=0.95  # 'mostly' typo
         )
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "query",
+    [
+        pytest.param("SELECT * FROM table", id="no batch"),
+        pytest.param("SELECT * FROM {{ batch }}", id="invalid batch syntax 1"),
+        pytest.param("SELECT * FROM {batch}", id="invalid batch syntax 2"),
+    ],
+)
+def test_sql_expectation_invalid_query_raises_error(query: str):
+    with pytest.raises(pydantic.ValidationError):
+        SqlExpectation(query=query)
+
+
+@pytest.fixture
+def taxi_db_path() -> str:
+    return file_relative_path(__file__, "../test_sets/quickstart/yellow_tripdata.db")
+
+
+@pytest.fixture
+def sqlite_datasource(
+    in_memory_runtime_context: AbstractDataContext, taxi_db_path: str
+) -> SqliteDatasource:
+    context = in_memory_runtime_context
+    datasource_name = "my_sqlite_datasource"
+    return context.sources.add_sqlite(
+        datasource_name, connection_string=f"sqlite:///{taxi_db_path}"
+    )
+
+
+@pytest.mark.sqlite
+def test_sql_expectation_validate(sqlite_datasource: SqliteDatasource):
+    datasource = sqlite_datasource
+    asset = datasource.add_table_asset("yellow_tripdata_sample_2022_01")
+
+    batch_request = asset.build_batch_request()
+    batch = asset.get_batch_list_from_batch_request(batch_request)[0]
+
+    expectation = SqlExpectation(
+        query="SELECT * FROM {active_batch} WHERE passenger_count > 6"
+    )
+    result = batch.validate(expectation)
+
+    # DB contains a single deviation (passenger_count == 7)
+    assert result.success is False
+
+    res = result.result
+    assert res["observed_value"] == 1
+
+    details = res["details"]
```

Comment:
nit: Spreading this out into 3 lines with variables feels like it makes this harder to read that just `res["details"]["value_counts"]["unexpected_rows"]`. Also asserting on what that unexpected row is would help folks reading the test to understand what the output actually is.